### PR TITLE
Introduces random string as label

### PR DESCRIPTION
**Pull request by Jonas-WMDE**
Since we have a working unique constraint over label/description tests
were failing, because of that. This patch fixes this by using a random
string as label.

### CODE REVIEW

File: tests/browser/features/step_definitions/label_steps.rb [hunk 1]
```diff
@@ -24,6 +24,14 @@
   end
 end
 
+When(/^I enter random string as label$/) do
+  on(ItemPage) do |page|
```
Comment (thiemowmde) [inline tests/browser/features/step_definitions/label_steps.rb]: It should be possible to call the method above, not introducing such an obvious code duplication.

File: tests/browser/features/step_definitions/label_steps.rb [hunk 2]
```diff
@@ -74,6 +88,14 @@
   end
 end
 
+Then(/^random string should be displayed as label$/) do
+  on(ItemPage) do |page|
```
Comment (thiemowmde) [inline tests/browser/features/step_definitions/label_steps.rb]: It should be possible to call the method above, not introducing such an obvious code duplication.

**Comment by thiemowmde** [comment]
+1.5. I wonder if we can avoid that code duplication.

**Comment by Jonas-WMDE** [comment]
Those are not functions and I don't have a clue how we can avoid the code duplication.

**Comment by thiemowmde** [comment]
https://github.com/cucumber/cucumber/wiki/Calling-Steps-from-Step-Definitions

**Comment by Jonas-WMDE** [comment]
So this doesn't work if you have any clue, just tell me..
When(/^I enter random string as label$/) do
  @random_string = generate_random_string(14)
  step "I enter #{@random_string} as label" 
end

**Comment by thiemowmde** [comment]
I miss quotes.

**Comment by JeroenDeDauw** [comment]
Can you not just use different strings? Randomness is something to avoid in tests, unless you are explicitly going for running them with different values and recording those such as done in fuzz testing. Having tests that work/fail some of the time is really not good. I don't know if that will happen here, though just adding random strings is scary.

**Comment by Jonas-WMDE** [comment]
We have a lot of random stuff in the browser tests. In this case it is necessary to use random labels, because the test items are not being deleted after running a scenario.

**Comment by Jonas-WMDE** [comment]
@thiemowmde  `step "I enter "#{@random_string}" as label"` this doesn't work either.

**Comment by thiemowmde** [comment]
`step "I enter \"#{@random_string}\" as label"`? o_O

**Comment by Jonas-WMDE** [comment]
BINGO!!

**Pull request merged**